firebase_admin/_utils.py

```diff
@@ -32,3 +34,28 @@ def _get_initialized_app(app):
 def get_app_service(app, name, initializer):
     app = _get_initialized_app(app)
     return app._get_service(name, initializer) # pylint: disable=protected-access
+
+def project_id(app):
+    """Returns the project ID associated with the given App.
+
+    This function first attempts to find the project ID from App options. If not present,
+    it attempts to get the project ID from the authentication credential used to initialize
+    the App. Finally, it attempts to get the project ID by referencing the GCLOUD_PROJECT
+    environment variable.
+
+    Args:
+      app: A Firebase App instance.
+
+    Returns:
+      str: A project ID value or None.
+    """
+    pid = app.options.get('projectId')
```

Comment:
Wouldn't it make sense to initialize app to get all the values from the environment as opposed doing this lookup every time someone calls project_id(app)?

Reply:
Done. Now this logic runs as part of `firebase_admin.initialize_app()`. Updated test cases accordingly.